### DISCUSSION TITLE
"There is no proper AP/PA data"

### DISCUSSION THREAD

**Discussion by Minarose** [McIntosh-Lab/tvb-ukbb#177] [Help] [answered]
Hello,

I have a reverse phase encoded image that I would like to use for TOPUP, however, there seems to be an issue with identifying the phase encoding direction. The file manager script identifies my data as "single-direction DWI". From my understanding of the [code](https://github.com/McIntosh-Lab/tvb-ukbb/blob/14324fe5b1ba2ea7b6e40a60ac7b56f9657691ae/bb_pipeline_tools/bb_file_manager.py#L441), it seems like it identifies the data based on whether it can find "PA" or "AP" in the nii.gz file name but I'm not sure. I tried adding that to the nii.gz file names but I still got the same error. Does it read this information from the image header? I'm using dwi acquired using a Phillips scanner which doesn't include a PhaseEncodingDirection tag in the json file or header. I would really appreciate it if you could advise me on how to get the file manager to recognize my AP/PA data. I have a single reverse phase encoded image in addition to the concatenated multi shell images.

Thank you,
Minarose

**Comment by shen4brains** [comment, marked as answer]
Hi Minarose,

The original UKBB pipeline supported AP/PA images for TOPUP. We haven't removed this feature nor altered any of that existing code, so theoretically the pipeline should still work for AP/PA images. However, we haven't worked with any data that required this feature so I cannot comment on whether it actually works. I also cannot comment on whether the pipeline requires the PhaseEncodingDirection tag in the JSON file. I'll note that the UKBB was collected exclusively on Siemens' scanners so there could very well be differences in image handling requirements.
